main.cpp

```diff
@@ -45,6 +52,44 @@ int main() {
                 .build();
     });
 
+    server.register_handler<HttpMethod::POST>("/search/{id, ip_address}", [](const HttpRequestContext &context) {
+        std::string player_id = context.path_params.at{"id"};
+        std::string player_address = context.path_params.at{"ip_address"};
+        PlayerInfo player(search_id, client_address);
+
+        if(!search_manager.check_for_active_search(player)){
+            return HttpResponse::new_builder().status(204).build();
+        } else {
+            std::pair<unsigned, PlayerInfo> active_searching_player = search_manager.get_player_info_with_search_id();
+            lobby_manager.create_lobby(player, active_searching_player.second, active_searching_player.first);
+            search_manager.erase_first_active_search();
+        };
+
+        return HttpResponse::new_builder()
```

Comment:
How would the client making the search request know what searchId the server has generated?